triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/TriggerRunner.scala

```diff
@@ -3,78 +3,85 @@
 
 package com.daml.lf.engine.trigger
 
-import akka.actor.typed.{ActorRef, Behavior, PostStop, Signal}
-import akka.actor.typed.scaladsl.AbstractBehavior
 import akka.actor.typed.SupervisorStrategy._
 import akka.actor.typed.scaladsl.Behaviors
-import akka.actor.typed.scaladsl.ActorContext
+import akka.actor.typed.{ActorRef, Behavior, PostStop}
 import akka.stream.Materializer
 import com.daml.grpc.adapter.ExecutionSequencerFactory
-import com.daml.logging.{ContextualizedLogger, LoggingContextOf}
-import LoggingContextOf.{label, newLoggingContext}
-import com.daml.scalautil.Statement.discard
+import com.daml.logging.LoggingContextOf.{label, newLoggingContext}
+import com.daml.logging.{ContextualizedLogger}
+import spray.json._
 
 class InitializationHalted(s: String) extends Exception(s) {}
 class InitializationException(s: String) extends Exception(s) {}
 
 object TriggerRunner {
+  private val logger = ContextualizedLogger.get(this.getClass)
+
   type Config = TriggerRunnerImpl.Config
 
   trait Message
   final case object Stop extends Message
+  final case class Status(replyTo: ActorRef[TriggerStatus]) extends Message
+
+  sealed trait TriggerStatus
+  final case object QueryingACS extends TriggerStatus
+  final case object Running extends TriggerStatus
+  final case object Stopped extends TriggerStatus
+
+  implicit val triggerStatusFormat: JsonFormat[TriggerStatus] = new JsonFormat[TriggerStatus] {
+    override def read(json: JsValue): TriggerStatus = json match {
+      case JsString("running") => Running
+      case JsString("stopped") => Stopped
+      case JsString("querying ACS") => QueryingACS
+      case _ => deserializationError(s"TriggerStatus expected")
+    }
+    override def write(obj: TriggerStatus): JsValue = obj match {
+      case QueryingACS => JsString("querying ACS")
+      case Running => JsString("running")
+      case Stopped => JsString("stopped")
+    }
+  }
 
   def apply(config: Config, name: String)(
       implicit esf: ExecutionSequencerFactory,
       mat: Materializer): Behavior[TriggerRunner.Message] =
     newLoggingContext(label[Config with Trigger], config.loggingExtension) {
       implicit loggingContext =>
-        Behaviors.setup(ctx => new TriggerRunner(ctx, config, name))
-    }
-}
-
-final class TriggerRunner private (
```

Comment:
I don’t think it’s strictly necessary to kill this class but I find mixing behaviors and classes to be even more confusing than sticking to just behaviors and I was tracking down some bug which went away when I switched it over :shrug: